spec/services/partnerships/report_spec.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+RSpec.describe Partnerships::Report do
+  let(:school) { create :school }
+  let(:cohort) { create :cohort }
+  let(:lead_provider) { create :lead_provider }
+  let(:delivery_partner) { create :delivery_partner }
+
+  subject(:service_instance) do
+    described_class.new(
+      school_id: school.id,
+      cohort_id: cohort.id,
+      lead_provider_id: lead_provider.id,
+      delivery_partner_id: delivery_partner.id,
+    )
+  end
+
+  subject(:result) { service_instance.call }
```

Comment:
I think we are conflating the execution of the call with the resulting object. Can we have something like `execute` from one of the other tests to run the service call and a `let` to retrieve the resulting partnership?

Reply:
Yup, rspec is pretty much noun-oriented, which is tricky when you're testing a verb... Sometimes I do:

```
subject(:execution) { -> { do_sth }`

it { is_expected.to change ...` }
```

Hmmmm, actually now that I think about this, we could make it work out of the box by defining `to_proc` method on the service. Then the service itself would act like a verb in the test while result can be defined as process output